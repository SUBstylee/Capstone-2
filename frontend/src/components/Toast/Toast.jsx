import { useState,useEffect } from 'react';
import { Button } from 'react-bootstrap';
import './Toast.scss';

const Toast = ({messageTitle, messageText, top, right, time}) => {
    const [seen,setSeen]=useState(false);

    useEffect(()=>{
        const toastTimer = setTimeout(()=>{
            setSeen(true);
        },`${time}`);
        return ()=>clearTimeout(toastTimer);
    },[time]);

  return (
    <div onClick={()=>setSeen(true)} className={`Toast ${seen?'hidden':''}`} style={{top: `${top}`, right: `${right}`}}>
        <Button variant='danger' className="btn-sm"><i className="fa-solid fa-x"></i></Button>
        <h4>{messageTitle}</h4>
        <p>{messageText}</p>
    </div>
  );
};

export default Toast;