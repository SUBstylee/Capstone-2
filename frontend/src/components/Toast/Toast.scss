.Toast{
    position: fixed;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    background-color: #fff;
    border: 3px solid #c40200;
    padding: 20px;
    border-radius: 8px;
    margin: 10px;
    z-index: 1;
    max-width: 300px;
    
    h4{
        display: inline-block;
        margin: 0 auto 10px;
        text-decoration: underline;
    }

    Button{
        position: absolute;
        top: -10px;
        right: -10px;
    }
}
.hidden{
    display: none;
}