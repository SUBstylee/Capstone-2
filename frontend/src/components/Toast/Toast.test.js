import { render } from '@testing-library/react';
import Toast from './Toast.jsx';
import store from '../../store';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

test("renders Toast component without crashing", () => {
    const { queryByText } = render(
        <Provider store={store}><BrowserRouter><Toast messageTitle='test title' messageText='test text' /></BrowserRouter></Provider>
    );

    expect(queryByText('test title')).toBeInTheDocument();
    expect(queryByText('test text')).toBeInTheDocument();
});

test("matches snapshot", function () {
    const { asFragment } = render(
        <Provider store={store}><BrowserRouter><Toast messageTitle='test title' messageText='test text' /></BrowserRouter></Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});