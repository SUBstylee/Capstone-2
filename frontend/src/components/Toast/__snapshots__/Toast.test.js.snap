// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Toast "
  >
    <button
      class="btn-sm btn btn-danger"
      type="button"
    >
      <i
        class="fa-solid fa-x"
      />
    </button>
    <h4>
      test title
    </h4>
    <p>
      test text
    </p>
  </div>
</DocumentFragment>
`;
